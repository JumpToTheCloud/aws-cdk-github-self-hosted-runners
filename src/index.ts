export * from './ec2/ec2-launch-template';
