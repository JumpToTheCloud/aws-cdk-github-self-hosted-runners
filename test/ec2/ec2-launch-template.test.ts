import { App, Stack } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { EC2LaunchTemplate } from '../../src';

let app: App;
let stack: Stack;

describe('Github Runner EC2 Launch Tempate', () => {
  beforeEach(() => {
    app = new App();
    stack = new Stack(app, 'TestStack');
  });

  it('launch Template validation test', () => {
    new EC2LaunchTemplate(stack, 'GithubRunnerEC2LaunchTemplate');

    const template = Template.fromStack(stack);

    expect(template.toJSON()).toMatchSnapshot();
  });

  it('should have One Launch Template', () => {
    new EC2LaunchTemplate(stack, 'GithubRunnerEC2LaunchTemplate');

    const template = Template.fromStack(stack);

    template.resourceCountIs('AWS::EC2::LaunchTemplate', 1);
  });
});
