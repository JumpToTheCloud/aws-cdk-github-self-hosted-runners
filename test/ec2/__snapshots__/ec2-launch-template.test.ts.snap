// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Github Runner EC2 Launch Tempate launch Template validation test 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "GithubRunnerEC2LaunchTemplateCF0FB830": {
      "Properties": {
        "LaunchTemplateData": {
          "TagSpecifications": [
            {
              "ResourceType": "instance",
              "Tags": [
                {
                  "Key": "Name",
                  "Value": "TestStack/GithubRunnerEC2LaunchTemplate/EC2LaunchTemplate",
                },
              ],
            },
            {
              "ResourceType": "volume",
              "Tags": [
                {
                  "Key": "Name",
                  "Value": "TestStack/GithubRunnerEC2LaunchTemplate/EC2LaunchTemplate",
                },
              ],
            },
          ],
        },
        "LaunchTemplateName": "github-self-hosted-runner-ec2-template",
        "TagSpecifications": [
          {
            "ResourceType": "launch-template",
            "Tags": [
              {
                "Key": "Name",
                "Value": "TestStack/GithubRunnerEC2LaunchTemplate/EC2LaunchTemplate",
              },
            ],
          },
        ],
      },
      "Type": "AWS::EC2::LaunchTemplate",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
